i6_models/parts/conformer/mhsa.py

```diff
@@ -60,3 +61,39 @@ def forward(self, input_tensor: torch.Tensor, sequence_mask: torch.Tensor) -> to
         output_tensor = torch.nn.functional.dropout(output_tensor, p=self.dropout, training=self.training)  # [B,T,F]
 
         return output_tensor
+
+
+class ConformerMHSAV2Config(ConformerMHSAV1Config):
+    pass
+
+
+class ConformerMHSAV2(torch.nn.Module):
+    """
+    Conformer multi-headed self-attention module
+    """
+
+    def __init__(self, cfg: ConformerMHSAV2Config):
+
+        super().__init__()
+
+        self.layernorm = torch.nn.LayerNorm(cfg.input_dim)
+        self.mhsa = MultiheadAttentionV1(cfg)
+        self.dropout = cfg.dropout
+
+    def forward(self, input_tensor: torch.Tensor, sequence_mask: torch.Tensor) -> torch.Tensor:
+        """
+        Apply layer norm and multi-head self attention and dropout
+
+        :param input_tensor: Input to the self attention of shape (B, T, F)
+        :param sequence_mask: bool mask of shape (B, T), True signals within sequence, False outside, will be inverted to match the torch.nn.MultiheadAttention module
+        which will be applied/added to dot product, used to mask padded key positions out
+        """
+        inv_sequence_mask = compat.logical_not(sequence_mask)
+        output_tensor = self.layernorm(input_tensor)  # [B,T,F]
```

Comment:
```suggestion
        output_tensor = self.layernorm(input_tensor)  # [B, T, F]
```
please also review other shape docs